src/impls/$zip/__tests__/$zip.test.ts

```diff
@@ -26,4 +26,27 @@ describe($`zip`, () => {
       ]);
     }),
   );
+
+  it(
+    'allows different source types',
+    $async(() => {
+      const twoSources = $zip($wrap([1, 2, 3]), $wrap(['a', 'b', 'c']));
+      expect($await($toArray(twoSources))).toEqual([
+        [1, 'a'],
+        [2, 'b'],
+        [3, 'c'],
+      ]);
+
+      const threeSources = $zip(
+        $wrap([1, 2, 3]),
+        $wrap(['a', 'b', 'c']),
+        $wrap([true, false, true]),
+      );
+      expect($await($toArray(threeSources))).toEqual([
+        [1, 'a', true],
```

Comment:
I think if you do `[1, 'a', true] as const` typescript will understand this as a tuple type which will ensure that this test won't pass  unless the types you added are present and working.

Reply:
I'm not familiar with using `as const` in this context, but I went ahead and added it for you.